PythonScript/src/NotepadPlusWrapper.cpp

```diff
@@ -591,7 +603,14 @@ BufferEncoding NotepadPlusWrapper::getEncoding()
 
 BufferEncoding NotepadPlusWrapper::getBufferEncoding(intptr_t bufferID)
 {
-	return static_cast<BufferEncoding>(callNotepad(NPPM_GETBUFFERENCODING, static_cast<WPARAM>(bufferID)));
+	if (bufferID >= 0)

```

Comment:
Do you have an example which shows the problem with bufferID < 0? Seems negative values are ok. The bufferID is directly used as pointer. There is just no check in N++ that the bufferID is a valid one and therefore an invalid pointer access could happen if the bufferID is unknown.

Reply:
For example notepad.reloadBuffer(-1,True) or notepad.getLangType(-1)

Reply:
Callstack is:
 
 notepad++.exe!Buffer::setDeferredReload() Zeile 465	C++
>	notepad++.exe!FileManager::reloadBufferDeferred(Buffer * id) Zeile 681	C++
 	notepad++.exe!Notepad_plus::doReload(Buffer * id, bool alert) Zeile 448	C++
 	notepad++.exe!Notepad_plus::process(HWND__ * hwnd, unsigned int message, unsigned __int64 wParam, __int64 lParam) Zeile 411	C++
 	notepad++.exe!Notepad_plus_Window::runProc(HWND__ * hwnd, unsigned int message, unsigned __int64 wParam, __int64 lParam) Zeile 120	C++
 	notepad++.exe!Notepad_plus_Window::Notepad_plus_Proc(HWND__ * hwnd, unsigned int message, unsigned __int64 wParam, __int64 lParam) Zeile 96	C++

and FileManager::reloadBufferDeferred(BufferID id) at
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/3a52ab1af16df8833eaab293106a85ba909be864/PowerEditor/src/ScitillaComponent/Buffer.cpp#L677

is not checking if the provided bufferID is a valid one. So I guess that needs to be done somewhere at the wrapper level. 
Similar to https://github.com/notepad-plus-plus/notepad-plus-plus/blob/3a52ab1af16df8833eaab293106a85ba909be864/PowerEditor/src/ScitillaComponent/Buffer.cpp#L506